src/benchmarklib/synthetic_table_generator.hpp

```diff
@@ -97,18 +97,21 @@ class SyntheticTableGenerator {
       return static_cast<T>(input);
     } else if constexpr (std::is_floating_point_v<T>) {
       return static_cast<T>(input) * 0.999999f;
-    } else {
+    } else if constexpr (std::is_same_v<T, pmr_string>) {
+      Assert(input >= 0, "Integer values need to be positive in order to be converted to a pmr_string.");
+
       constexpr auto generated_string_length = size_t{10};
       constexpr auto prefix_length = size_t{4};
       constexpr auto variable_string_length = generated_string_length - prefix_length;
 
-      Assert(input >= 0, "Integer values need to be positive in order to be converted to a pmr_string.");
-
       const std::vector<char> chars = {'0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'A', 'B', 'C', 'D', 'E', 'F',
                                        'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V',
                                        'W', 'X', 'Y', 'Z', 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l',
                                        'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z'};
       const size_t chars_base = chars.size();
+
+      // For the current settings, integer valures are guaranteed to not exceed the value range. However, the assertion
+      // is kept as the generated string length might change or the interface (e.g., taking size_t input).
```

Comment:
1) `int` does not guarantee you anything. `int32_t` would.
2) What does this comment explain (read: what information would be missing without it)?

Reply:
It's supposed to say that the following assert should be kept, even though it can be never be wrong at the moment. We could remove it, but I am afraid the next user who changes the input to something larger will stumble upon this issue. So I'd keep the assertion.

Reply:
Maybe just "Implicitly guaranteed by int32_t parameter, but asserting this just in case the parameter changes in the future"?